tfx/extensions/google_cloud_kubernetes/runner.py

```diff
@@ -0,0 +1,245 @@
+                        ),
+                    ],
+                    ports=[
+                        client.V1ContainerPort(
+                            container_port=5000,
+                        ),
+                    ],
+                    resources=client.V1ResourceRequirements(
+                        limits={
+                            'nvidia.com/gpu': num_gpus_per_worker,
+                        },
+                    ) if num_gpus_per_worker > 0 else None,
+                ),
+            ],
+            restart_policy=kube_utils.RestartPolicy.NEVER.value,
+        ),
+    )
+    try:
+      api_instance.create_namespaced_pod(namespace='default', body=pod)
+    except ApiException as e:
+      logging.error(
+          'Exception when calling CoreV1Api->create_namespaced_pod: %s' % e)
+
+  logging.info('created {} worker pods'.format(num_workers))
+
+
+def create_worker_services(training_inputs: Dict[Text, Any],
+                           unique_id: Text):
+  """Create worker services for multi-worker training."""
+  num_workers = training_inputs.get('num_workers', 1)
+  service_names = _build_service_names(num_workers=num_workers,
+                                       unique_id=unique_id)
+  api_instance = kube_utils.make_core_v1_api()
+
+  # TODO(ericlege): consider using a jinja2 template instead
+  for i in range(num_workers):
+    service = client.V1Service(
+        metadata=client.V1ObjectMeta(
+            name=service_names[i],
+        ),
+        spec=client.V1ServiceSpec(
+            selector={
+                'name': 'training',
+                'id': unique_id,
+                'task': str(i),
+            },
+            ports=[
+                client.V1ServicePort(
+                    port=5000,
+                ),
+            ],
+        ),
+    )
+    try:
+      api_instance.create_namespaced_service(namespace='default', body=service)
+    except ApiException as e:
+      logging.error(
+          'Exception when calling CoreV1Api->create_namespaced_service: %s' % e)
+  logging.info('created {} worker services'.format(num_workers))
+
+
+def delete_worker_services(training_inputs: Dict[Text, Any],
+                           unique_id: Text):
+  """Clean up worker services deployed to the kubernetes cluster."""
+  num_workers = training_inputs.get('num_workers', 1)
+  service_names = _build_service_names(num_workers=num_workers,
+                                       unique_id=unique_id)
+  api_instance = kube_utils.make_core_v1_api()
+  for service_name in service_names:
+    try:
+      api_instance.delete_namespaced_service(namespace='default',
+                                             name=service_name)
+    except ApiException as e:
+      logging.error(
+          'Exception when calling CoreV1Api->delete_namespaced_service: %s' % e)
+  logging.info('Deleted {} worker services'.format(num_workers))
+
+
+def start_gke_training(input_dict: Dict[Text, List[types.Artifact]],
```

Comment:
Let's name this more neutrally -- "start_kubernetes_training"?

Reply:
Renamed.